spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarAdministrationBadContextTests.java

```diff
@@ -0,0 +1,41 @@
+package org.springframework.pulsar.core;
+
+import static org.assertj.core.api.Assertions.assertThatIllegalStateException;
+import java.util.Map;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.junit.jupiter.api.Test;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+
+public class PulsarAdministrationBadContextTests extends AbstractContainerBaseTests {
+
+		@Test
+		void testDecrementingPartitionCount() {
```

Comment:
I see why you pulled this out to its own test as its a bit diff model and it kills the context during load. I think we can pull this back into the other test by doing the following restructuring:

1. Instead of pre-create a large one and then try to create a smaller one at load time we can create a large one during load time then in the test method call PulsarAdministration.createOrModifyTopics w/ the smaller topic and assert failure at that point. Something like this: 
```java
	@Nested
	@ContextConfiguration(classes = DecrementPartitionCount.DecrementPartitionCountConfig.class)
	class DecrementPartitionCount {

		@Test
		void testDecrementingPartitionCount(@Autowired ObjectProvider<PulsarTopic> expectedTopics) throws Exception {
			assertThatTopicsExist(expectedTopics.stream().toList());
			PulsarTopic smallerTopic = PulsarTopic.builder("dpc-partitioned-1").setNumberOfPartitions(4).build();
			assertThatIllegalStateException()
					.isThrownBy(() -> pulsarAdministration.createOrModifyTopics(smallerTopic))
					.withMessage("Topic persistent://public/default/dpc-partitioned-1 found with 8 partitions. Needs to be deleted first.");
		}

		@Configuration(proxyBeanMethods = false)
		static class DecrementPartitionCountConfig {
			@Bean
			PulsarTopic partitionedTopic() {
				return PulsarTopic.builder("dpc-partitioned-1").setNumberOfPartitions(8).build();
			}
		}
	}
```
wdyt?

Reply:
Yes, initially I wanted to have all of them in the same place but failed to do so with the way I had the test structured. It's great that we can have it all in one place now!